libcnb-data/src/layer_content_metadata.rs

```diff
@@ -62,11 +61,11 @@ pub struct LayerContentMetadata<M> {
     pub metadata: M,
 }
 
-impl Default for LayerContentMetadata<GenericMetadata> {
+impl Default for LayerContentMetadata<Option<toml::Value>> {
     fn default() -> Self {
         Self {
             types: LayerContentTypeTable::default(),
-            metadata: GenericMetadata::default(),
+            metadata: Default::default(),
```

Comment:
Clippy pedantic says:

```
$ cargo clippy --all-targets --all-features -- --deny warnings
    Checking libcnb-data v0.3.0 (/Users/emorley/src/libcnb.rs/libcnb-data)
error: calling `std::option::Option::default()` is more clear than this expression
  --> libcnb-data/src/layer_content_metadata.rs:68:23
   |
68 |             metadata: Default::default(),
   |                       ^^^^^^^^^^^^^^^^^^ help: try: `std::option::Option::default()`
   |
   = note: `-D clippy::default-trait-access` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_trait_access
```

Reply:
I changed this for now to unblock this PR and we can have this discussion later.

Reply:
I'm totally fine with disabling pedantic lints that we disagree with - they are in the pedantic category after all :-) 

The reason I enabled the pedantic lints was so that we could get the advantage of the 80% of them that are useful, with the understanding that it's ok to disable subsets as we see fit.